botorch/optim/optimize.py

```diff
@@ -707,6 +744,105 @@ def _gen_batch_initial_conditions_local_search(
     raise RuntimeError(f"Failed to generate at least {min_points} initial conditions")
 
 
+def _validate_constraints(
+    bounds: Tensor,
+    inequality_constraints: Optional[List[Tuple[Tensor, Tensor, float]]] = None,
+    equality_constraints: Optional[List[Tuple[Tensor, Tensor, float]]] = None,
+) -> None:
+    r"""Validate constraints for acquisition function optimization.
+
+    Checks that the constraints define a bounded, non-empty polytope.
+
+    Args:
+        bounds: A `2 x d` tensor of lower and upper bounds for each column of `X`.
+            If there are no box constraints, bounds should be an empty `0 x d`-dim
+            tensor.
+        inequality constraints: A list of tuples (indices, coefficients, rhs),
+            with each tuple encoding an inequality constraint of the form
+            `\sum_i (X[indices[i]] * coefficients[i]) >= rhs`
+        equality constraints: A list of tuples (indices, coefficients, rhs),
+            with each tuple encoding an inequality constraint of the form
+            `\sum_i (X[indices[i]] * coefficients[i]) = rhs`
+    """
+    # We solve the following Linear Program to ensure that he constraint set
+    # is non-empty and bounded:
+    #
+    #   max_x |x|_1
+    #     s.t. bounds(x)
+    #          inequality_constraints(x)
+    #          equality_constraints(x)
+    #
+    # To do this we can introduce auxiliary variables s and solve the
+    # following standard formulation:
+    #
+    #   min_(x, s) - sum_i(s_i)
+    #     s.t. -x <= s <= x
```

Comment:
I don't think this is correct - we're trying to solve
```
max \sum_i |x_i|
```
and the correct transformation would be 
```
max \sum_i s_i
s_i <= |x_i|
```
The constraint `|x_i| >= s_i` is equal to `(x_i >= s_i) OR (x_i <= -s_i)` and is not convex and so won't be representable as an LP.

The constraint given here, `-x _i<= s_i <= x_i` is equivalent to `|s_i| <= x`, which is convex but is not correct for the problem we're trying to solve. In particular it will only provide the correct answer for positive values of `x_i`; if the inequality and equality constraints are unbounded towards -Inf but has volume in the positive orthant, this LP will not detect the unboundedness.